packages/page-staking/src/Performance/FinalityCommittee.tsx

```diff
@@ -1,23 +1,71 @@
 // Copyright 2017-2025 @polkadot/app-staking authors & contributors
 // SPDX-License-Identifier: Apache-2.0
 
-import React from 'react';
+import React, { useMemo, useRef, useState } from 'react';
 
-import { AddressSmall, CardSummary, SummaryBox, Table } from '@polkadot/react-components';
+import { CardSummary, SummaryBox, Table } from '@polkadot/react-components';
+import { useNextTick } from '@polkadot/react-hooks';
 
+import Filtering from '../Filtering.js';
 import { useTranslation } from '../translate.js';
+import FinalizerAddress from './FinalizerAddress/FinalizerAddress.js';
+import Legend from './Legend.js';
+import useAbftScores from './useAbftScores.js';
 import { useFinalityCommittee } from './useFinalityCommittee.js';
 
 interface Props {
+  className?: string;
   session: number;
   currentSession: number;
 }
 
-function FinalityCommittee ({ currentSession, session }: Props) {
+export interface Finalizer {
+  accountId: string;
+  abftScore?: number;
+}
+
+function FinalityCommittee ({ className, currentSession, session }: Props) {
   const { t } = useTranslation();
   const finalityCommitteeAddresses = useFinalityCommittee(session, currentSession);
+  const { abftScores, scoresEnabled } = useAbftScores(session);
+  const isNextTick = useNextTick();
+  const [nameFilter, setNameFilter] = useState<string>('');
+
+  const finalizers: Finalizer[] = useMemo(() => {
+    if (finalityCommitteeAddresses?.length) {
+      return finalityCommitteeAddresses.map((accountId, index) => {
+        if (abftScores?.points.length === finalityCommitteeAddresses.length) {
```

Comment:
On what occasions the lengths of these 2 arrays would be different?

Reply:
Okay, either way is fine by me. Just note that the lengths comparison check might be useful if there's _any_ chance of the lengths not matching, which IIUC would cause the `finalizers` array to contain inaccurate data with `abftScore` and `accountId` not matching.